### PR TITLE
Add icdf for Beta, Gamma, and StudentT

### LINKED ISSUE #1365: icdf of beta distribution is not implemented
```python
>>> import numpyro.distributions as dist
>>> d = dist.Beta(1, 5)
>>> d.icdf(0.975)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ayaka/venv/lib/python3.10/site-packages/numpyro/distributions/distribution.py", line 463, in icdf
    raise NotImplementedError
NotImplementedError
```

NumPyro version: 0.9.1

### PR DESCRIPTION
**Pull request by fehiepsi**
Fixes #1365

### CODE REVIEW

File: numpyro/distributions/util.py [hunk 1]
```diff
@@ -414,6 +414,37 @@ def clamp_probs(probs):
     return jnp.clip(probs, a_min=finfo.tiny, a_max=1.0 - finfo.eps)
 
 
+def betainc(a, b, x):
+    try:
+        from tensorflow_probability.substrates.jax.math import betainc as betainc_fn
+    except ImportError:
```
Comment (martinjankowiak) [inline numpyro/distributions/util.py:420]: do you want to include an error message here as below?

**Review by martinjankowiak — COMMENTED** [review]
are all the cdfs and icdfs here compared against scipy in the tests?

File: numpyro/distributions/util.py (hunk 1 above)
Comment (fehiepsi) [reply numpyro/distributions/util.py:420]: Here we use the default jax implementation (which does not have grad over all params)

**Comment by fehiepsi** [comment]
Yes, those cdf/icdf are compared against scipy in the tests

**Review by martinjankowiak — APPROVED** [review]

**Pull request merged**